cpp/src/arrow/io/test_common.cc

```diff
@@ -48,6 +52,28 @@ void AssertFileContents(const std::string& path, const std::string& contents) {
 
 bool FileExists(const std::string& path) { return std::ifstream(path.c_str()).good(); }
 
+Status PurgeLocalFileFromOsCache(const std::string& path) {
+#ifdef _WIN32
+  return Status::NotImplemented("Cannot yet purge files from cache on Windows");
+#else
+  int fd = open(path.c_str(), O_WRONLY);
+  if (fd < 0) {
+    return IOErrorFromErrno(errno, "open on ", path,
+                            " to clear from cache did not succeed.");
+  }
+  int err = posix_fadvise(fd, 0, 0, POSIX_FADV_DONTNEED);
```

Comment:
Ah, it also seems this isn't supported on MacOS: https://github.com/apache/arrow/runs/4811739968?check_suite_focus=true#step:7:779

Reply:
Good catch.  I looked at file.cc and realized we are using:

```
#if defined(POSIX_FADV_WILLNEED)
```
which seems much better.  I can't use that #if in the benchmark itself though (unless I want to import fcntl.h which wouldn't be the worst thing) so I changed it to SkipWithError.